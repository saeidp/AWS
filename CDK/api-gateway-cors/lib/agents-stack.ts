import * as cdk from 'aws-cdk-lib';
import { CfnOutput, Stack, StackProps } from 'aws-cdk-lib';
import { AppConfig, createResourceName } from '@assembly-line-core/common';
import { Construct } from 'constructs';
// import { AgentLookupData } from './constructs/agent-lookup-data';
import { AgentsConstruct } from './constructs/agents-construct';

export interface AgentLookupDataStackProps extends StackProps {
  readonly appConfig: AppConfig;
}

export class AgentsStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: AgentLookupDataStackProps) {
    super(scope, id, props);

    if (!props?.env) {
      throw Error('props.env is required');
    }

    if (!props?.env.region) {
      throw Error('props.env.region is required');
    }

    if (!props?.env.account) {
      throw Error('props.env.account is required');
    }

    const { appConfig } = props;

    // const agentLookupData = new AgentLookupData(this, 'agent-lookup-data', {
    //   appConfig: props.appConfig
    // });

    // new CfnOutput(this, 'RestApiUrl', {
    //   value: agentLookupData.restApi.url
    // });
   
    const agents = new AgentsConstruct(this, 'agents', {
      appConfig: props.appConfig,
    });

    new CfnOutput(this, 'RestApiUrl', {
      value: agents.restApi.url
    })   
  }
}
