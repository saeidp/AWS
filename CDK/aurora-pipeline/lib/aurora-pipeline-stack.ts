import * as cdk from '@aws-cdk/core';
import { Artifact, IStage, Pipeline } from "@aws-cdk/aws-codepipeline";
import { CloudFormationCreateUpdateStackAction, CodeBuildAction, CodeStarConnectionsSourceAction } from "@aws-cdk/aws-codepipeline-actions";
import { BuildSpec, LinuxBuildImage, PipelineProject } from "@aws-cdk/aws-codebuild";
import { AuroraStack } from './aurora-stack';
export class AuroraPipelineStack extends cdk.Stack {
  private readonly pipeline: Pipeline;
  private readonly auroraPipelineSourceOutput: Artifact;
  private readonly auroraPipelineBuildOutput: Artifact;
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    this.pipeline = new Pipeline(this, "AuroraPipeline", {
      pipelineName: "AuroraPipeline",
      crossAccountKeys: true,
      restartExecutionOnUpdate: true
    });

    this.auroraPipelineSourceOutput = new Artifact("AuroraPipelineSourceOutput");
    this.auroraPipelineBuildOutput = new Artifact("AuroraPipelineBuildOutput");

    this.pipeline.addStage({
      stageName: "Bitbucket_Source",
      actions: [
        new CodeStarConnectionsSourceAction({
          actionName: 'Pipeline_Source',
          owner: 'data-mesh-foundation',
          repo: 'aurora-pipeline',
          branch: "master",
          output: this.auroraPipelineSourceOutput,
          connectionArn: 'arn:aws:codestar-connections:ap-southeast-2:447421689278:connection/b3a45c41-32b7-4453-96b6-d17e77a9462b',

        }),
      ]
    });
    this.pipeline.addStage({
      stageName: "Build",
      actions: [
        new CodeBuildAction({
          actionName: "Pipeline_Build",
          input: this.auroraPipelineSourceOutput,
          outputs: [this.auroraPipelineBuildOutput],
          project: new PipelineProject(this, "CdkBuildProject", {
            environment: {
              buildImage: LinuxBuildImage.STANDARD_5_0,
            },
            buildSpec: BuildSpec.fromSourceFilename(
              "build-specs/aurora-pipeline-build-spec.yml"
            ),
          }),
        }),
      ],
    });

    this.pipeline.addStage({
      stageName: "Pipeline_Update",
      actions: [
        new CloudFormationCreateUpdateStackAction({
          actionName: "Pipeline_Update",
          stackName: "AuroraPipelineStack",
          templatePath: this.auroraPipelineBuildOutput.atPath("AuroraPipelineStack.template.json"),
          adminPermissions: true,
        }),
      ],
    });

  }

  public addAuroraStage(auroraStack: AuroraStack, stageName: string): IStage {
    return this.pipeline.addStage({
      stageName: stageName,
      actions: [
        new CloudFormationCreateUpdateStackAction({
          actionName: "Aurora_Update",
          account: auroraStack.account,
          region: auroraStack.region,
          stackName: auroraStack.stackName,
          templatePath: this.auroraPipelineBuildOutput.atPath(`${auroraStack.stackName}.template.json`),
          adminPermissions: true,
        })
      ]

    });
  }

}
