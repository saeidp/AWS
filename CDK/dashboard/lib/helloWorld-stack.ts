import * as cdk from '@aws-cdk/core';
import * as lambda from '@aws-cdk/aws-lambda-nodejs';
import { Runtime } from '@aws-cdk/aws-lambda';
import * as path from 'path';

export class HelloWorldStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        const helloWorld = new lambda.NodejsFunction(this, 'HelloWorldLambda', {
            runtime: Runtime.NODEJS_14_X,
            entry: path.join(__dirname, '..', 'api', 'hello-world', 'index.ts'),
            functionName: 'HelloWorld',
            handler: 'helloWorld',
        });

    }
}
