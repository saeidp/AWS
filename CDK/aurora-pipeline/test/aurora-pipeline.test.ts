import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as Pipeline from '../lib/aurora-pipeline-stack';

test('Pipeline Stack', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new Pipeline.AuroraPipelineStack(app, 'MyTestStack');
  // THEN
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();

});
