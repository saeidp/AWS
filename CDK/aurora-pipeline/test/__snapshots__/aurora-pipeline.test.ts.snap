// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline Stack 1`] = `
Object {
  "Resources": Object {
    "AuroraPipeline1B6DAAD3": Object {
      "DependsOn": Array [
        "AuroraPipelineRoleDefaultPolicy67270253",
        "AuroraPipelineRole7DE3AE80",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "EncryptionKey": Object {
            "Id": Object {
              "Fn::GetAtt": Array [
                "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                "Arn",
              ],
            },
            "Type": "KMS",
          },
          "Location": Object {
            "Ref": "AuroraPipelineArtifactsBucket562B6A6B",
          },
          "Type": "S3",
        },
        "Name": "AuroraPipeline",
        "RestartExecutionOnUpdate": true,
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "AuroraPipelineRole7DE3AE80",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "AWS",
                  "Provider": "CodeStarSourceConnection",
                  "Version": "1",
                },
                "Configuration": Object {
                  "BranchName": "master",
                  "ConnectionArn": "arn:aws:codestar-connections:ap-southeast-2:447421689278:connection/b3a45c41-32b7-4453-96b6-d17e77a9462b",
                  "FullRepositoryId": "data-mesh-foundation/aurora-pipeline",
                },
                "Name": "Pipeline_Source",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "AuroraPipelineSourceOutput",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRole9E8747FB",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Bitbucket_Source",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ProjectName": Object {
                    "Ref": "CdkBuildProject9382C38D",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "AuroraPipelineSourceOutput",
                  },
                ],
                "Name": "Pipeline_Build",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "AuroraPipelineBuildOutput",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineBuildPipelineBuildCodePipelineActionRole9AAD7CF1",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "CREATE_UPDATE",
                  "Capabilities": "CAPABILITY_NAMED_IAM",
                  "RoleArn": Object {
                    "Fn::GetAtt": Array [
                      "AuroraPipelinePipelineUpdateRoleB85ED3A6",
                      "Arn",
                    ],
                  },
                  "StackName": "AuroraPipelineStack",
                  "TemplatePath": "AuroraPipelineBuildOutput::AuroraPipelineStack.template.json",
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "AuroraPipelineBuildOutput",
                  },
                ],
                "Name": "Pipeline_Update",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelinePipelineUpdateCodePipelineActionRoleC78C5AF9",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Pipeline_Update",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "AuroraPipelineArtifactsBucket562B6A6B": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "KMSMasterKeyID": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineRole7DE3AE80",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRole9E8747FB",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CdkBuildProjectRoleE0B6FEB0",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CdkBuildProjectRoleE0B6FEB0",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelinePipelineUpdateRoleB85ED3A6",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelinePipelineUpdateCodePipelineActionRoleC78C5AF9",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Delete",
    },
    "AuroraPipelineArtifactsBucketEncryptionKeyAlias087F67CE": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AliasName": "alias/codepipeline-myteststackaurorapipeline09d6d7a4",
        "TargetKeyId": Object {
          "Fn::GetAtt": Array [
            "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
            "Arn",
          ],
        },
      },
      "Type": "AWS::KMS::Alias",
      "UpdateReplacePolicy": "Delete",
    },
    "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRole9E8747FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRoleDefaultPolicyE3F39BCC": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "codestar-connections:UseConnection",
              "Effect": "Allow",
              "Resource": "arn:aws:codestar-connections:ap-southeast-2:447421689278:connection/b3a45c41-32b7-4453-96b6-d17e77a9462b",
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineArtifactsBucket562B6A6B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "AuroraPipelineArtifactsBucket562B6A6B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "s3:PutObjectAcl",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "AuroraPipelineArtifactsBucket562B6A6B",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRoleDefaultPolicyE3F39BCC",
        "Roles": Array [
          Object {
            "Ref": "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRole9E8747FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "AuroraPipelineBuildPipelineBuildCodePipelineActionRole9AAD7CF1": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "AuroraPipelineBuildPipelineBuildCodePipelineActionRoleDefaultPolicy90522AC6": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CdkBuildProject9382C38D",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "AuroraPipelineBuildPipelineBuildCodePipelineActionRoleDefaultPolicy90522AC6",
        "Roles": Array [
          Object {
            "Ref": "AuroraPipelineBuildPipelineBuildCodePipelineActionRole9AAD7CF1",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "AuroraPipelinePipelineUpdateCodePipelineActionRoleC78C5AF9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "AuroraPipelinePipelineUpdateCodePipelineActionRoleDefaultPolicy8A096758": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "iam:PassRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelinePipelineUpdateRoleB85ED3A6",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineArtifactsBucket562B6A6B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "AuroraPipelineArtifactsBucket562B6A6B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "cloudformation:CreateStack",
                "cloudformation:DescribeStack*",
                "cloudformation:GetStackPolicy",
                "cloudformation:GetTemplate*",
                "cloudformation:SetStackPolicy",
                "cloudformation:UpdateStack",
                "cloudformation:ValidateTemplate",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":cloudformation:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":stack/AuroraPipelineStack/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "AuroraPipelinePipelineUpdateCodePipelineActionRoleDefaultPolicy8A096758",
        "Roles": Array [
          Object {
            "Ref": "AuroraPipelinePipelineUpdateCodePipelineActionRoleC78C5AF9",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "AuroraPipelinePipelineUpdateRoleB85ED3A6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudformation.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "AuroraPipelinePipelineUpdateRoleDefaultPolicy2EADE9CE": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineArtifactsBucket562B6A6B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "AuroraPipelineArtifactsBucket562B6A6B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "*",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "AuroraPipelinePipelineUpdateRoleDefaultPolicy2EADE9CE",
        "Roles": Array [
          Object {
            "Ref": "AuroraPipelinePipelineUpdateRoleB85ED3A6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "AuroraPipelineRole7DE3AE80": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "AuroraPipelineRoleDefaultPolicy67270253": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineArtifactsBucket562B6A6B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "AuroraPipelineArtifactsBucket562B6A6B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineBitbucketSourcePipelineSourceCodePipelineActionRole9E8747FB",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineBuildPipelineBuildCodePipelineActionRole9AAD7CF1",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelinePipelineUpdateCodePipelineActionRoleC78C5AF9",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "AuroraPipelineRoleDefaultPolicy67270253",
        "Roles": Array [
          Object {
            "Ref": "AuroraPipelineRole7DE3AE80",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkBuildProject9382C38D": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "EncryptionKey": Object {
          "Fn::GetAtt": Array [
            "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
            "Arn",
          ],
        },
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "CdkBuildProjectRoleE0B6FEB0",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "build-specs/aurora-pipeline-build-spec.yml",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "CdkBuildProjectRoleDefaultPolicy3C7ECB00": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "CdkBuildProject9382C38D",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "CdkBuildProject9382C38D",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "CdkBuildProject9382C38D",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "AuroraPipelineArtifactsBucket562B6A6B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "AuroraPipelineArtifactsBucket562B6A6B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "AuroraPipelineArtifactsBucketEncryptionKey4B78DB3E",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkBuildProjectRoleDefaultPolicy3C7ECB00",
        "Roles": Array [
          Object {
            "Ref": "CdkBuildProjectRoleE0B6FEB0",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkBuildProjectRoleE0B6FEB0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
